runtime/src/launch/slurm-srun/launch-slurm-srun.c

```diff
@@ -224,14 +249,30 @@ static char* chpl_launch_create_command(int argc, char* argv[],
     if (account && strlen(account) > 0) {
       fprintf(slurmFile, "#SBATCH --account=%s\n", account);
     }
- 
-    // set the output name to either the user specified
-    // or to the binaryName.<jobID>.out if none specified
-    if (outputfn!=NULL) {
-      fprintf(slurmFile, "#SBATCH --output=%s\n", outputfn);
+
+    // set the output file name to either the user specified
+    // name or to the binaryName.<jobID>.out if none specified
+    if (outputfn != NULL) {
+      sprintf(stdoutFile,      "%s", outputfn);
+      sprintf(stdoutFileNoFmt, "%s", outputfn);
     }
     else {
-      fprintf(slurmFile, "#SBATCH --output=%s.%%j.out\n", argv[0]);
+      sprintf(stdoutFile,      "%s.%s.out", argv[0], "%j");
+      sprintf(stdoutFileNoFmt, "%s.%s.out", argv[0], "$SLURM_JOB_ID");
+    }
+
+    // If we're buffering the output, set the temp output file name.
+    // It's always /tmp/binaryName.<jobID>.out.
+    if (bufferStdout != NULL) {
+      sprintf(tmpStdoutFile,      "/tmp/%s.%s.out", argv[0], "%j");
```

Comment:
I think it's usually a better practice to use `$TMPDIR` var, then `P_tmpdir` macro, and default to `/tmp`. I know we use `/tmp` in most places, though, so no need to do that here. Just thought I'd mention it.